bug: The instance is frequently restarted

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```sql
2022-12-30T05:44:04.278997Z mysqld_safe mysqld restarted
2022-12-30T05:44:07.318860Z mysqld_safe Number of processes running now: 0
2022-12-30T05:44:07.330660Z mysqld_safe mysqld restarted
2022-12-30T05:44:10.237983Z mysqld_safe Number of processes running now: 0
2022-12-30T05:44:10.250383Z mysqld_safe mysqld restarted
```
###tianmu.log
```sql
[2022-12-30 13:41:10.718119] [22162] [WARN] [exception.cpp:42] MSG: Exception: assert failed on i < m_idx.size() at tianmu_attr.h:382, msg: [bad dpn index 111/1].

STACK TRACE BEGIN
        /stonedb57/install//bin/mysqld(Tianmu::common::AssertException::AssertException(char const*, char const*, int, std::string const&)+0x1b7) [0x2f87007]
        /stonedb57/install//bin/mysqld(Tianmu::core::TianmuAttr::get_dpn(unsigned long)+0xe2) [0x2d3f18a]
        /stonedb57/install//bin/mysqld(Tianmu::core::TianmuAttr::LockPackForUse(unsigned int)+0x50) [0x2d361e4]
        /stonedb57/install//bin/mysqld(_ZSt13__invoke_implIvRMN6Tianmu4core10TianmuAttrEFvjERPS2_IRiEET_St21__invoke_memfun_derefOT0_OT1_DpOT2_+0x7e) [0x2d84c39]
        /stonedb57/install//bin/mysqld(std::__invoke_result<void (Tianmu::core::TianmuAttr::*&)(unsigned int)<Tianmu::core::TianmuAttr*&, int&> >::type std::__invoke<void (Tianmu::core::TianmuAttr::*&)(unsigned int), Tianmu::core::TianmuAttr*&, int&>(void (Tianmu::core::TianmuAttr::*&)(unsigned int), Tianmu::core::TianmuAttr*&, int&)+0x57) [0x2d83725]
        /stonedb57/install//bin/mysqld(void std::_Bind<void (Tianmu::core::TianmuAttr::*(Tianmu::core::TianmuAttr*, int))(unsigned int)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>)+0x75) [0x2d82087]
        /stonedb57/install//bin/mysqld(void std::_Bind<void (Tianmu::core::TianmuAttr::*(Tianmu::core::TianmuAttr*, int))(unsigned int)>::operator()<, void>()+0x2a) [0x2d7f8b2]
        /stonedb57/install//bin/mysqld(std::_Function_handler<void (), std::_Bind<void (Tianmu::core::TianmuAttr::*(Tianmu::core::TianmuAttr*, int))(unsigned int)> >::_M_invoke(std::_Any_data const&)+0x20) [0x2d7c336]
        /stonedb57/install//bin/mysqld(std::function<void ()>::operator()() const+0x32) [0x2c5d0a6]
        /stonedb57/install//bin/mysqld(Tianmu::core::FunctionExecutor::FunctionExecutor(std::function<void ()>, std::function<void ()>)+0x71) [0x2c94e29]
        /stonedb57/install//bin/mysqld(Tianmu::core::DataPackLock::DataPackLock(Tianmu::core::TianmuAttr*, int const&)+0xcf) [0x2d788a1]
        /stonedb57/install//bin/mysqld(std::_MakeUniq<Tianmu::core::DataPackLock>::__single_object std::make_unique<Tianmu::core::DataPackLock, Tianmu::core::TianmuAttr*&, int>(Tianmu::core::TianmuAttr*&, int&&)+0x51) [0x2d7b4ac]
        /stonedb57/install//bin/mysqld(Tianmu::core::TianmuTable::Iterator::LockPacks()+0xe8) [0x2d6f6ee]
        /stonedb57/install//bin/mysqld(Tianmu::core::TianmuTable::Iterator::FetchValues()+0x2b) [0x2d6f509]
        /stonedb57/install//bin/mysqld(Tianmu::core::TianmuTable::Iterator::GetData(int)+0x1b) [0x2da1775]
        /stonedb57/install//bin/mysqld(Tianmu::handler::ha_tianmu::fill_row(unsigned char*)+0x13e) [0x2d98b62]
        /stonedb57/install//bin/mysqld(Tianmu::handler::ha_tianmu::rnd_pos(unsigned char*, unsigned char*)+0x389) [0x2d97db5]
        /stonedb57/install//bin/mysqld(handler::ha_rnd_pos(unsigned char*, unsigned char*)+0x125) [0x1d6c15b]
        /stonedb57/install//bin/mysqld(handler::rnd_pos_by_record(unsigned char*)+0xac) [0x1d7ac08]
        /stonedb57/install//bin/mysqld(Rows_log_event::do_index_scan_and_update(Relay_log_info const*)+0x378) [0x26444fa]
        /stonedb57/install//bin/mysqld(Rows_log_event::do_apply_event(Relay_log_info const*)+0x10ef) [0x26469a5]
        /stonedb57/install//bin/mysqld(Log_event::apply_event(Relay_log_info*)+0x1b3) [0x262e67d]
        /stonedb57/install//bin/mysqld(apply_event_and_update_pos(Log_event**, THD*, Relay_log_info*)+0x422) [0x26a2dc9]
        /stonedb57/install//bin/mysqld() [0x26a4463]
        /stonedb57/install//bin/mysqld(handle_slave_sql+0xf84) [0x26aadf4]
        /stonedb57/install//bin/mysqld(pfs_spawn_thread+0x170) [0x2bc8c61]
        /lib64/libpthread.so.0(+0x81a3) [0x7f52bca0e1a3]
        /lib64/libc.so.6(clone+0x6d) [0x7f52b9e9dfad]
STACK TRACE END

[2022-12-30 13:41:10.718507] [22162] [ERROR] [ha_tianmu.cpp:1193] MSG: An exception is caught: assert failed on i < m_idx.size() at tianmu_attr.h:382, msg: [bad dpn index 111/1]
[2022-12-30 13:41:10.718734] [22162] [ERROR] [MYSQL:0] MSG: Slave SQL for channel '': Could not execute Delete_rows event on table zz.sbtest1; assert failed on i < m_idx.size() at tianmu_attr.h:382, msg: [bad dpn index 111/1], Error_code: 6; Can't find record in 'sbtest1', Error_code: 1032; handler error HA_ERR_END_OF_FILE; the event's master log mybinlog.000012, end_log_pos 300383577, Error_code: 6
[2022-12-30 13:41:10.718821] [22162] [WARN] [MYSQL:0] MSG: Slave: assert failed on i < m_idx.size() at tianmu_attr.h:382, msg: [bad dpn index 111/1] Error_code: 6
[2022-12-30 13:41:10.718863] [22162] [WARN] [MYSQL:0] MSG: Slave: Can't find record in 'sbtest1' Error_code: 1032
[2022-12-30 13:41:10.722166] [22162] [ERROR] [MYSQL:0] MSG: Error running query, slave SQL thread aborted. Fix the problem, and restart the slave SQL thread with "SLAVE START". We stopped at log 'mybinlog.000012' position 300383162.
[2022-12-30 13:41:10.722271] [22162] [ERROR] [engine.cpp:763] MSG: Roll back query '(null)'
[2022-12-30 13:41:10.722311] [22162] [INFO] [transaction.cpp:90] MSG: rollback transaction 63ae79f300002711.
[2022-12-30 13:41:10.722372] [22162] [INFO] [tianmu_table.cpp:393] MSG: roll back table zz../zz/sbtest1.tianmu
[2022-12-30 13:41:10.722518] [22162] [ERROR] [engine.cpp:763] MSG: Roll back query '(null)'
```

### Expected behavior

_No response_

### How To Reproduce

_No response_

### Environment

./mysqld  Ver 5.7.36-StoneDB-debug for Linux on x86_64 (build-)
build information as follow:
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: 26b011f
        Last commit time: Date:   Thu Dec 29 10:16:32 2022 +0800
        Build time: Date: Fri Dec 30 11:30:38 CST 2022

### Are you interested in submitting a PR to solve the problem?

- [X] Yes, I will!

--- a/mysql-test/suite/tianmu/r/issue1174.result
+++ b/mysql-test/suite/tianmu/r/issue1174.result
@@ -1,0 +1,303 @@
+use test;
+drop table if exists `ttt`;
+drop table if exists `t3`;
+include/master-slave.inc
+[connection master]
+#
+# Has primary key
+# 
+[on master]
+create table ttt(id int primary key,name varchar(64))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on master]
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on master]
+drop table ttt;
+CREATE TABLE t3 (a int not null, b int not null, primary key (a,b))engine=innodb;
+insert into t3 values (1,1),(2,1),(1,3),(4,5),(6,7),(8,9),(0,0);
+delete from t3 where a in (4,6,1,0);
+select * from t3 order by a desc;
+a	b
+8	9
+2	1
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from t3 order by a desc;
+a	b
+8	9
+2	1
+[on master]
+drop table t3;
+include/sync_slave_sql_with_master.inc
+#
+# Have unique constraints
+# 
+[on master]
+create table ttt(id int NOT NULL UNIQUE,name varchar(64))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on master]
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on master]
+drop table ttt;
+CREATE TABLE t3 (a int not null, b int not null, UNIQUE(a,b))engine=innodb;
+insert into t3 values (1,1),(2,1),(1,3),(4,5),(6,7),(8,9),(0,0);
+delete from t3 where a in (4,6,1,0);
+select * from t3 order by a desc;
+a	b
+8	9
+2	1
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from t3 order by a desc;
+a	b
+8	9
+2	1
+[on master]
+drop table t3;
+include/sync_slave_sql_with_master.inc
+#
+# With secondary index
+# 
+[on master]
+create table ttt(id int not null,name varchar(64),index(id))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on master]
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on master]
+drop table ttt;
+CREATE TABLE t3 (a int not null, b int not null, index(a,b))engine=innodb;
+insert into t3 values (1,1),(2,1),(1,3),(4,5),(6,7),(8,9),(0,0);
+delete from t3 where a in (4,6,1,0);
+select * from t3 order by a desc;
+a	b
+8	9
+2	1
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from t3 order by a desc;
+a	b
+8	9
+2	1
+[on master]
+drop table t3;
+include/sync_slave_sql_with_master.inc
+#
+# common
+# 
+[on master]
+create table ttt(id int , name varchar(64))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+id	name
+3	lllll
+2	lllll
+1	lllll
+0	lllll
+[on master]
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on slave]
+include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+id	name
+111	你好
+88	hhhb
+31	hhhb
+11	aaa
+6	hhhb
+4	hhhb
+3	lllll
+2	lllll
+[on master]
+drop table ttt;
+include/sync_slave_sql_with_master.inc
+stop slave;

--- a/mysql-test/suite/tianmu/t/issue1174.test
+++ b/mysql-test/suite/tianmu/t/issue1174.test
@@ -1,0 +1,200 @@
+-- source include/have_tianmu.inc
+
+use test;
+--disable_warnings
+drop table if exists `ttt`;
+drop table if exists `t3`;
+--enable_warnings
+
+--disable_warnings
+-- source include/master-slave.inc
+--enable_warnings
+
+
+--echo #
+--echo # Has primary key
+--echo # 
+--echo [on master]
+connection master;
+create table ttt(id int primary key,name varchar(64))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on master]
+connection master;
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+--echo [on master]
+connection master;
+drop table ttt;
+CREATE TABLE t3 (a int not null, b int not null, primary key (a,b))engine=innodb;
+insert into t3 values (1,1),(2,1),(1,3),(4,5),(6,7),(8,9),(0,0);
+delete from t3 where a in (4,6,1,0);
+select * from t3 order by a desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from t3 order by a desc;
+--echo [on master]
+connection master;
+drop table t3;
+--source include/sync_slave_sql_with_master.inc
+
+--echo #
+--echo # Have unique constraints
+--echo # 
+--echo [on master]
+connection master;
+create table ttt(id int NOT NULL UNIQUE,name varchar(64))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on master]
+connection master;
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+--echo [on master]
+connection master;
+drop table ttt;
+CREATE TABLE t3 (a int not null, b int not null, UNIQUE(a,b))engine=innodb;
+insert into t3 values (1,1),(2,1),(1,3),(4,5),(6,7),(8,9),(0,0);
+delete from t3 where a in (4,6,1,0);
+select * from t3 order by a desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from t3 order by a desc;
+--echo [on master]
+connection master;
+drop table t3;
+--source include/sync_slave_sql_with_master.inc
+
+--echo #
+--echo # With secondary index
+--echo # 
+--echo [on master]
+connection master;
+create table ttt(id int not null,name varchar(64),index(id))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on master]
+connection master;
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+--echo [on master]
+connection master;
+drop table ttt;
+CREATE TABLE t3 (a int not null, b int not null, index(a,b))engine=innodb;
+insert into t3 values (1,1),(2,1),(1,3),(4,5),(6,7),(8,9),(0,0);
+delete from t3 where a in (4,6,1,0);
+select * from t3 order by a desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from t3 order by a desc;
+--echo [on master]
+connection master;
+drop table t3;
+--source include/sync_slave_sql_with_master.inc
+
+--echo #
+--echo # common
+--echo # 
+--echo [on master]
+connection master;
+create table ttt(id int , name varchar(64))engine=innodb;
+insert into ttt values(111,'你好');
+insert into ttt values(11,'aaa');
+insert into ttt values(3,'hhhb');
+insert into ttt values(2,'hhhb');
+insert into ttt values(1,'hhhb');
+insert into ttt values(0,'hhhb');
+insert into ttt values(31,'hhhb');
+insert into ttt values(32,'zzz');
+insert into ttt values(4,'hhhb');
+insert into ttt values(5,'zzz');
+insert into ttt values(6,'hhhb');
+insert into ttt values(7,'zzz');
+insert into ttt values(88,'hhhb');
+insert into ttt values(98,'zzz');
+update ttt set name='lllll' where id in(1,3,2,0);
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt where id in(1,3,2,0) order by id desc;
+--echo [on master]
+connection master;
+delete from ttt where id=1;
+delete from ttt where id=0;
+delete from ttt where name='zzz';
+select * from ttt order by id desc;
+--echo [on slave]
+--source include/sync_slave_sql_with_master.inc
+select * from ttt order by id desc;
+--echo [on master]
+connection master;
+drop table ttt;
+--source include/sync_slave_sql_with_master.inc
+
+stop slave;

--- a/storage/tianmu/handler/ha_tianmu.cpp
+++ b/storage/tianmu/handler/ha_tianmu.cpp
@@ -1165,7 +1165,7 @@ int ha_tianmu::rnd_pos(uchar *buf, uchar *pos) {
 
   int ret = HA_ERR_END_OF_FILE;
   try {
-    if (table->in_use->slave_thread && table->s->primary_key != MAX_INDEXES && pos && pos[0] != '\0') {
+    if (table->in_use->slave_thread && table->s->primary_key != MAX_INDEXES) {
       ret = index_read(buf, pos, ref_length, HA_READ_KEY_EXACT);
     } else {
       uint64_t position = my_get_ptr(pos, ref_length);

